src/server.rs

```diff
@@ -143,17 +143,18 @@ impl Server {
     ///
     /// Returns an error if `other_total_shares.len()` is not equal to this
     //// server's `dimension`.
-    pub fn merge_total_shares(&mut self, other_total_shares: &[Field]) -> Result<(), ServerError> {
+    pub fn merge_total_shares(&mut self, other_total_shares: &[F]) -> Result<(), ServerError> {
         Ok(merge_vector(&mut self.accumulator, other_total_shares)?)
     }
 
     /// Choose a random point for polynomial evaluation
     ///
     /// The point returned is not one of the roots used for polynomial
     /// evaluation.
-    pub fn choose_eval_at(&self) -> Field {
+    pub fn choose_eval_at(&self) -> F {
+        let mut rng = rand::thread_rng();
         loop {
-            let eval_at = Field::from(rand::random::<u32>());
+            let eval_at = F::rand(&mut rng);
```

Comment:
It seems like everywhere `F::rand` is used, the caller has to provide a `rand::Rng`, which is a bit of a hassle. I see the value in exposing a variant that lets the caller provide an RNG, but could we perhaps have two variants, one that takes no arguments and defaults to `rand::thread_rng()` and another that lets the caller provide a `rand::Rng`?

Reply:
Hmm, well already have  `random_field_from_seed` which maps a seed to a field element. The client and aggregators need to agree on how to do this procedure. This is basically doing the same thing as the current `fn rand<R: Rng + ?Sized>(rng: &mut R) -> Self` API, which is a little messy. I think we should just nix `fn rand<R: Rng + ?Sized>(rng: &mut R) -> Self;`  call and just have `fn rand() -> Self`.